libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -120,6 +120,12 @@ bool ImmutableValidator::visit(WhileStatement const& _whileStatement)
 	return false;
 }
 
+void ImmutableValidator::endVisit(UserDefinedTypeName const& _userDefinedTypeName)
+{
+	if (auto const callableDef = dynamic_cast<CallableDeclaration const*>(_userDefinedTypeName.annotation().referencedDeclaration))
+		visitCallableIfNew(callableDef->resolveVirtual(m_currentContract));
```

Comment:
I'm not sure if we should do virtual lookup here.

Reply:
Well, this is exactly the same thing as we're doing when visiting the identifiers which is what was previously used for the super constructor calls. If we shouldn't do this here then we shouldn't do it there either.

Reply:
It is different because we do not do virtual lookup for functions that use path notation either. The idea behind is that this is always qualified access: `a.x` means "the x defined in a" while `x` means "whatever x currently is overwritten by"

Reply:
Mind you, this is also for modifiers, which can have virtual behavior (if I recall correctly)

Reply:
In this case though, the AST type that was previously `Identifier` changed to `UserDefinedTypeName`, so even though it might just be a single identifier, it still is of the UserDefinedTypeName type. 

We can add a check to see if it is actually a path or just a single word and behave differently based on that...